### PR TITLE
FileManager: Fix linkItem(atPath:toPath) for directories and symlinks

### PR DESCRIPTION
**Pull request by spevans**
- Recurse into any directories found in the source.

- Copy any symlinks that are found. Only regular files are hard linked.

### CODE REVIEW

**Comment by spevans** [comment]
@swift-ci please test

**Comment by spevans** [comment]
@swift-ci please test

**Comment by spevans** [comment]
@swift-ci please test

File: Foundation/FileManager.swift [hunk 1]
```diff
@@ -530,22 +530,14 @@ open class FileManager : NSObject {
         }
     }
 
-    open func copyItem(atPath srcPath: String, toPath dstPath: String) throws {
+    private func _processItem(atPath srcPath: String, toPath dstPath: String, _ body: (String, String, FileAttributeType) throws -> ()) throws {
```
Comment (kperryua) [inline Foundation/FileManager.swift]: This method seems too generically named for something that creates directories. The rest looks great though.

**Comment by spevans** [comment]
@swift-ci please test and merge

**Comment by spevans** [comment]
@swift-ci please test and merge

**Pull request merged**